### ISSUE TITLE
Add EXT2/3/4 FS support to NuttX

### ISSUE THREAD

**Issue by acassis** [apache/nuttx#12415]
I found a nice project that could be integrated on NuttX to support ext2, ext3 and ext4:

https://github.com/gkostka/lwext4

To get it BSD only we need to remove the extended attribute support as explained in the readme.

**Comment by anoraktrend** [comment]
They might want to reimplement it, instead.

**Comment by acassis** [comment]
Hi @anoraktrend nice you see you here! I remember browsing your git repository sometime ago.

**Comment by anoraktrend** [comment]
> Hi @anoraktrend nice you see you here! I remember browsing your git repository sometime ago.

Howdy! I was looking at this as a potential alternative kernel for Iglunix, (frankensrc, etc), because it's a neat project. ext4 is something that is something of a requirement, so this issue is a good one for that reason.

**Comment by acassis** [comment]
Amazing @anoraktrend welcome aboard! We are glad you enjoyed it!

**Comment by acassis** [comment]
I didn't know about Iglunix, but used many small Linux distros in the past like Slitaz and many others.
Does Iglunix use Toybox? Marco Casaroli is just porting it to NuttX (shell and some commands are already working), I think it could be something useful for you!

**Comment by anoraktrend** [comment]
> I didn't know about Iglunix, but used many small Linux distros in the past like Slitaz and many others. Does Iglunix use Toybox? Marco Casaroli is just porting it to NuttX (shell and some commands are already working), I think it could be something useful for you!

Yep, toybox, busybox, mksh, and the LLVM suite. We could replace some of busybox & toybox with @chimera-linux's chimerautils (a neat thing, it ports a lot of the FreeBSD userland to Linux).

**Comment by acassis** [comment]
Interesting! Some chimerautils tools seems too big for a small microcontroller, but some tools could be ported/reused on NuttX. Also I noticed that some tools are open Linux specific /proc and /sys files that only exist on Linux. These tools need to be modified as well.

**Comment by xiaoxiang781216** [comment]
we are porting https://github.com/gkostka/lwext4 to nuttx
